tests/e2e/dsc_creation_test.go

```diff
@@ -161,13 +161,9 @@ func (tc *testContext) testAllApplicationCreation(t *testing.T) error { //nolint
 		t.Parallel()
 		err = tc.testApplicationCreation(&(tc.testDsc.Spec.Components.Dashboard))
 		if tc.testDsc.Spec.Components.Dashboard.ManagementState == operatorv1.Managed {
-			if err != nil {
```

Comment:
I repeat my question, why do you push these changes in this PR? It's unrelated to adding the component.

Reply:
Adding the model registry tests pushed the cyclomatic complexity of the code beyond the linter limit. So I had to clean up un-necessary code like`if nil` check before calling `require.NoError`. It won't pass the linter without simplifying the obviously bad code. 
The tests for all components for `Disabled` state used some useless code that does something that will never be true (`if err == nil` check before calling `require.NoError`). It looked like it should be checking for an error instead, which seems to pass for other components, except CodeFlare and ModelMeshServing. So I modified the test code to do what it was doing before, i.e. use `err == nil` with `require.NoError`.

Reply:
I went back to the git history for this code and at one point the test code flipped from looking for errors for disabled components, to not doing anything at all. I'm changing it back to what it should be doing for disabled components, i.e. using `require.Error`.

Reply:
I added your tests to the incubation branch and `make lint` runs just fine. Probably you can also disable it with annotations if it causes problems with integration.

I did not claim that the tests are correct, but fixing them sounds like a separate task, not part of adding a component.

Reply:
Ok, taking my words back :) Updated golangci-lint to 1.55.2 and it fails, but actually in many more places.

Reply:
Here is your tests in github https://github.com/opendatahub-io/opendatahub-operator/pull/815